typescript/lib/config.ts

```diff
@@ -822,5 +823,29 @@ export class AIConfigRuntime implements AIConfig {
     return this.metadata.models?.[modelName];
   }
 
+  /**
+   * Generates a ModelMetadata object based on the provided inference settings and model ID.
+   *
+   * @param inferenceSettings - The inference settings to be used for the model.
+   * @param modelId - The unique identifier for the model.
+   * @returns A ModelMetadata object that includes the model's name and optional settings.
+   */
+  public generateModelMetadata(
+    inferenceSettings: InferenceSettings,
+    modelId: string
+  ) {
+    const overrideSettings = extractOverrideSettings(
+      this,
+      inferenceSettings,
+      modelId
+    );
+
+    if (!overrideSettings) {
```

Comment:
if (!overrideSettings || overrideSettings === {})

Reply:
Since overrideSettings is an object, javascript won't let me compare to {} with `=== {}`. Updated to `Object.keys(overrideSettings).length === 0`